### DISCUSSION TITLE
How to analyse samples with previously-generated spectral library

### DISCUSSION THREAD

**Discussion by adi772** [vdemichev/DiaNN#1022] [Q&A]
Hi Vadim,
I analysed a batch of ~40 samples some time ago - I will hereby refer to this as analysis1 and batch1. An in-silico spectral library was generated while analysing batch 1. At the time, I supplied FASTA file and enabled: (i) Generate spectral library; (ii) FASTA digest for library generation; (iii) Deep learning-based spectra etc, and (iv) MBR. The analysis was successful (results as expected). According to the analysis1 log, it takes ~6.5 hours to generate this library. 

I now wish to re-analyse another batch of samples (batch2), very similar in nature, with the library generated from analysis1. I used the relevant .speclib file, queued the samples, unticked the aforementioned options. However, I repeatedly received output that looks like this:

> 0:00] Spectral library loaded: 1961 protein isoforms, 1874 protein groups and 7427 precursors in 6141 elution groups.
[0:00] Loading FASTA E:\fasta files\UP000005640_9606_human.fasta
[2:25] Reannotating library precursors with information from the FASTA database
[2:25] 7427 precursors generated
[2:25] Gene names missing for some isoforms
[2:25] Library contains 1961 proteins, and 1950 genes
[2:25] Initialising library
[2:25] File #1/9
[2:25] Loading run <file_path_removed_from_Github_post>
[2:30] 7427 library precursors are potentially detectable
[2:30] Processing...
[2:31] Cannot perform MS1 mass calibration, too few confidently identified precursors
[2:31] RT window set to 34.6874
[2:31] Peak width: 0
[2:31] Scan window radius set to 12
[2:31] Cannot perform mass calibration, too few confidently identified precursors
[2:31] Cannot perform MS1 mass calibration, too few confidently identified precursors
[2:32] Removing low confidence identifications
[2:32] Searching PTM decoys
[2:32] Removing interfering precursors
[2:32] Too few confident identifications, neural networks will not be used
[2:32] Number of IDs at 0.01 FDR: 0
[2:32] Number of IDs at 0.01 FDR: 0
[2:32] Calculating protein q-values
[2:32] Number of proteins identified at 1% FDR: 0 (precursor-level), 0 (protein-level) (inference performed using proteotypic peptides only)

In essence, it seems that although the library loads in fine, it cannot detect precursors among the samples. This is even the case when I run the same .raw files that were used to generate the library in the first place: sample set 1 (and at that time, 500+ proteins were detected).

I have tried the following:

1. Enabling/disabling "Use existing .quant files where available" 
2. Ensuring all parameters are identical to those used when generating the library
3. Including FASTA and enabling "Re-annotate". This is definitely the same FASTA used for analysis1.
4. Using .dia files rather than .raw files
5. Using .tsv version of the spectral library instead of .speclib
6. Fixing mass and MS1 accuracy (using values based on log from analysis1)
7. Using a different spectral library (also .speclib) generated from some similar, but mock samples prior to this (batch0).

In all cases, all samples return "Number of proteins identified at 1% FDR: 0 (precursor-level), 0 (protein-level) (inference performed using proteotypic peptides only".

My goal is to analyse batch2 using the library from analysis1. But now I cannot even reproduce results using batch1. Am I configuring the program incorrectly somehow? 

Thank you kindly for any help whatsoever.

**Comment by adi772** [comment]
Update: I was able to reproduce the issue from scratch. Essentially, I have analysed 9 samples _de novo_ and created a spectral library as described above for analysis1. 
[33_rerun.log.txt](https://github.com/vdemichev/DiaNN/files/15450666/33_rerun.log.txt)

Then, I have tried to re-analyse using .quant files from the above, along with the .speclib file generated above, but got the error:
_ERROR: a .quant file was obtained using a different spectral library / different library-free search settings_

```
Thread number set to 24
Output will be filtered at 0.01 FDR
Precursor/protein x samples expression level matrices will be saved along with the main report
Only peaks with correlation sum exceeding 2 will be considered
Peaks with correlation sum below 1 from maximum will not be considered
A single score will be used until RT alignment to save memory; this can potentially lead to slower search
Fast algorithm based on MS1 feature extraction for quicker library-free search will be applied; this significantly worsens the identification performance
Maximum number of variable modifications set to 5
Modification UniMod:35 with mass delta 15.9949 at M will be considered as variable
Modification UniMod:1 with mass delta 42.0106 at *n will be considered as variable
Existing .quant files will be used
When generating a spectral library, in silico predicted spectra will be retained if deemed more reliable than experimental ones
Implicit protein grouping: protein names; this determines which peptides are considered 'proteotypic' and thus affects protein FDR calculation
Fixed-width center of each elution peak will be used for quantification
Interference removal from fragment elution curves disabled
DIA-NN will optimise the mass accuracy automatically using the first run in the experiment. This is useful primarily for quick initial analyses, when it is not yet known which mass accuracy setting works best for a particular acquisition scheme.
The following variable modifications will be scored: UniMod:1 
DIA-NN exited
Unless the spectral library specified was created by this version of DIA-NN, it's strongly recommended to specify a FASTA database and use the 'Reannotate' function to allow DIA-NN to identify peptides which can originate from the N/C terminus of the protein: otherwise site localisation might not work properly for modifications of the protein N-terminus or for modifications which do not allow enzymatic cleavage after the modified residue
```


```
[0:00] Loading spectral library <file path deleted for github post> 33_rerun.tsv.speclib
[0:00] Library annotated with sequence database(s): E:\fasta files\UP000005640_9606_human.fasta
[0:00] Gene names missing for some isoforms
[0:00] Library contains 1616 proteins, and 1607 genes
[0:00] Spectral library loaded: 1616 protein isoforms, 1473 protein groups and 5466 precursors in 4476 elution groups.
[0:00] Initialising library

[0:00] Cross-run analysis
[0:00] Reading quantification information: 9 files
ERROR: a .quant file was obtained using a different spectral library / different library-free search settings
```

So, instead of using .quant files, I unticked that option, and tried again with .raw files - returning once again the output:

```Number of proteins identified at 1% FDR: 0 (precursor-level), 0 (protein-level) (inference performed using proteotypic peptides only) for all samples.``` 
(log attached below)

[33_rerun_repeat.log.txt](https://github.com/vdemichev/DiaNN/files/15450725/33_rerun_repeat.log.txt)

The bottleneck (in terms of run-time) seems to be processing the .raw files - the running log is dominated by 'Unknown token', most probably because it is Boxcar DIA (as you and I have [discussed before](https://github.com/vdemichev/DiaNN/discussions/482)) though this doesn't appear in the final log. So, I tried with converted .dia files - run finishes MUCH quicker (less than 1 minute rather than 48 minutes), but got the same error (log attached)
[33_rerun_repeat_dia_filetype.log.txt](https://github.com/vdemichev/DiaNN/files/15450755/33_rerun_repeat_dia_filetype.log.txt)

Essentially, it seems I am unable to successfully re-use spectral libraries generated in DIA-NN and it seems I must produce them _de novo_ every time. Am I doing something wrong?

Any help would be hugely appreciated!

**Comment by vdemichev** [comment]
Hi,

> Then, I have tried to re-analyse using .quant files from the above, along with the .speclib file generated above, but got the error:
ERROR: a .quant file was obtained using a different spectral library / different library-free search settings

That's expected behaviour by DIA-NN. The .quant files were generated with 33_rerun.predicted.speclib in this case and were saved only for the first MBR pass.

> So, instead of using .quant files, I unticked that option, and tried again with .raw files - returning once again the output:

There are a lot of non-default options in that log. Please try with all (literally all) options default, using 33_rerun.tsv (not speclib) and with --mass-acc-cal 100, as well as Log level set to 3.

Best,
Vadim
